packages/query-plan-executor/src/tracing/handler.test.ts

```diff
@@ -0,0 +1,226 @@
+      return handler.runInChildSpan('async-operation', async (_span) => {
+        await timers.setTimeout(10) // Simulate async work
+        return 'async-result'
+      })
+    })
+
+    // Verify the result
+    expect(result).toEqual('async-result')
+
+    // Verify a span was collected
+    expect(collector.spans.length).toEqual(1)
+    expect(collector.spans[0].name).toEqual('prisma:engine:async-operation')
+  })
+
+  test('runs without active collector', () => {
+    const handler = new TracingHandler(getTestTracer())
+
+    // This should not throw, even though there's no collector
+    const result = handler.runInChildSpan('test-operation', (_span) => {
+      return 'test-result'
+    })
+
+    // Verify the function executed normally
+    expect(result).toEqual('test-result')
+  })
+
+  test('handles errors in sync functions', () => {
+    const collector = TracingCollector.newInCurrentContext()
+    const handler = new TracingHandler(getTestTracer())
+
+    let error: Error | undefined
+    try {
+      tracingCollectorContext.withActive(collector, () => {
+        handler.runInChildSpan('error-operation', () => {
+          throw new Error('Test error')
+        })
+      })
+    } catch (e) {
+      error = e as Error
+    }
+
+    // Verify the error was propagated
+    expect(error).toBeDefined()
+    expect(error!.message).toEqual('Test error')
+
+    // Verify a span was collected despite the error
+    expect(collector.spans.length).toEqual(1)
+    expect(collector.spans[0].name).toEqual('prisma:engine:error-operation')
+  })
+
+  test('handles errors in async functions', async () => {
+    const collector = TracingCollector.newInCurrentContext()
+    const handler = new TracingHandler(getTestTracer())
+
+    let error: Error | undefined
+    try {
+      await tracingCollectorContext.withActive(collector, () => {
+        return handler.runInChildSpan('async-error-operation', async () => {
+          await timers.setTimeout(10) // Simulate async work
+          throw new Error('Async test error')
+        })
+      })
+    } catch (e) {
+      error = e as Error
+    }
+
+    // Verify the error was propagated
+    expect(error).toBeDefined()
+    expect(error!.message).toEqual('Async test error')
+
+    // Verify a span was collected despite the error
+    expect(collector.spans.length).toEqual(1)
+    expect(collector.spans[0].name).toEqual('prisma:engine:async-error-operation')
+  })
+
+  test('nested spans create parent-child relationship', () => {
+    const collector = TracingCollector.newInCurrentContext()
+    const handler = new TracingHandler(getTestTracer())
+
+    tracingCollectorContext.withActive(collector, () => {
+      handler.runInChildSpan('parent-operation', (_parentSpan) => {
+        // Run a child span inside the parent span
+        handler.runInChildSpan('child-operation', (_childSpan) => {
+          // Do some work in the child span
+        })
+      })
+    })
```

Comment:
Is this intentionally empty?

Reply:
yes, we just need the spans themselves, we don't care what happens within them or how long they take